From fb1995f58644ff55e6ea4f3ebb56bf807cad480b Mon Sep 17 00:00:00 2001
From: Apekshit <apeksharma@gmail.com>
Date: Tue, 10 May 2016 19:11:51 -0700
Subject: [PATCH] HBASE-13532 Make UnknownScannerException less scary by giving
 more information in the exception string. (Apekshit)

Signed-off-by: Matteo Bertozzi <matteo.bertozzi@cloudera.com>
Amending-Author: Andrew Purtell <apurtell@apache.org>
---
 .../java/org/apache/hadoop/hbase/regionserver/HRegionServer.java | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java
index 072ca2b..a66776c 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java
@@ -3220,9 +3220,14 @@ public class HRegionServer implements ClientProtos.ClientService.BlockingInterfa
       if (request.hasScannerId()) {
         rsh = scanners.get(scannerName);
         if (rsh == null) {
-          LOG.info("Client tried to access missing scanner " + scannerName);
+          LOG.warn("Client tried to access missing scanner " + scannerName);
           throw new UnknownScannerException(
-            "Name: " + scannerName + ", already closed?");
+            "Unknown scanner '" + scannerName + "'. This can happen due to any of the following "
+                + "reasons: a) Scanner id given is wrong, b) Scanner lease expired because of "
+                + "long wait between consecutive client checkins, c) Server may be closing down, "
+                + "d) RegionServer restart during upgrade.\nIf the issue is due to reason (b), a "
+                + "possible fix would be increasing the value of"
+                + "'hbase.client.scanner.timeout.period' configuration.");
         }
         scanner = rsh.s;
         HRegionInfo hri = scanner.getRegionInfo();
-- 
2.7.4

